board_ramps.h

```diff
@@ -58,22 +58,35 @@
 // 128x64 full graphics controller
 #define BEEPER             37
 
+#ifdef USING_12864_WITH_ENCODER_INTERRUPTS
+// these alternate pins might help - see https://www.marginallyclever.com/forums/topic/using-reprap-kit-from-amazon-ramps-1-6-lcd-display/#post-22131
+// these settings worked with my Mega 2560/Ramps 1.4 and are needed for using encoder interrupts since the interrupts use D18 and D19 
+#define LCD_PINS_RS        16
+#define LCD_PINS_ENABLE    17
+#define LCD_PINS_D4        23
+#else
+
 #define LCD_PINS_RS        19
 #define LCD_PINS_ENABLE    42
 #define LCD_PINS_D4        18
-// these alternate pins might help - see https://www.marginallyclever.com/forums/topic/using-reprap-kit-from-amazon-ramps-1-6-lcd-display/#post-22131
-//#define LCD_PINS_RS        16
-//#define LCD_PINS_ENABLE    17
-//#define LCD_PINS_D4        23
+#endif
 
 #define LCD_PINS_D5        25
 #define LCD_PINS_D6        27
 #define LCD_PINS_D7        29
 
 // Encoder rotation values
-#define BTN_EN1            31
-#define BTN_EN2            33
+#ifdef USING_12864_WITH_ENCODER_INTERRUPTS
+#define BTN_EN1            18    // solder a jumper from D33 on the display connector to D18 on the RAMPS 1.4 board
```

Comment:
Why are we soldering here?  the _old goes low when the old en1 would go high, right?  I'd rather define an _ACTIVE as LOW or HIGH and instead of (digitalRead(BTN_EN1) == LOW) use (digitalRead(BTN_EN1) == BTN_ACTIVE)

Reply:
The ramps 1.4 "shield" board layout plugs into the D31 and D33 pins.  Those will still be connected, but don't have interrupts, so I have solder a jumper wires (no header pin I can get to) and then connect the jumpers to D18 and D19 that have interrupts.  The D31 and D33 pins will still be electrically connected, so they are still set to inputs, but the their pull-ups aren't enabled.

As for the == LOW logic... that was the way the existing code did the logic so I used it to be consistent.  The logic level didn't change from before, just using different pins with interrupts.

Reply:
![PXL_20210104_001033538](https://user-images.githubusercontent.com/7231870/103492326-184f2400-4df8-11eb-9019-62f244e34cbf.jpg)